### DISCUSSION TITLE
Web-Api: How to read Power of each panel (jq usage problem)

### DISCUSSION THREAD

**Discussion by capiman** [tbnobody/OpenDTU#593] [General]
There is the following example in the documentation:

~$ curl --no-progress-meter http://192.168.10.10/api/livedata/status | jq '.total | .Power.v'
140.7999878

What do I need to change to read Power of each panel? What I already tried:

$ curl --no-progress-meter http://192.168.179.38/api/livedata/status | jq '.inverters | .[0].Power.v'
null
$ curl --no-progress-meter http://192.168.179.38/api/livedata/status | jq '.inverters | .[1].Power.v'
null
$ curl --no-progress-meter http://192.168.179.38/api/livedata/status | jq '.inverters | .Power.v'
jq: error (at <stdin>:0): Cannot index array with string "Power"
$ curl --no-progress-meter http://192.168.179.38/api/livedata/status | jq '.inverters | .serial'
jq: error (at <stdin>:0): Cannot index array with string "serial"
$

**Comment by jaui** [comment]
for inverter 0 use  
`jq '.inverters[0] | .name'`
`jq '.inverters[0] | .AC."0".Power.v'`
inverter 1 is read with
`jq '.inverters[1] | .name'`
`jq '.inverters[1] | .AC."0".Power.v'`

For each panel, i.e. 2 panels on inverter 0:
`jq '.inverters[0] | .DC."0".Power.v'`
`jq '.inverters[0] | .DC."1".Power.v'`

if curl doesn't support --no-progress-meter option, use -s (silent) option
`$ curl -s http://opendtu/api/livedata/status | jq '.total | .Power.v'`

**Comment by toledobastos** [comment]
Has anyone managed to read individual panel data using curl? I just need the field .YieldDay to collect the individual panel output over time, but nothing I tried returns individual panel data. This is of course odd because the web interface does show this data.

This doesn't work:
$ curl -s IP/api/livedata/status | jq '.inverters[0] | .DC."0".Power.v'
null

This works, but it's just the aggregate and not individual panel data:
$ curl -s IP/api/livedata/status | jq '.total | .YieldDay.v'

**Comment by francwalter** [comment]
So this is still unsolved?
I spent a while to search for the grid values (actual power of each of my four grids) but didnt find anything. So I found this discussion.
As you say, the web interface shows those data but where is it in the api?
